### DISCUSSION TITLE
Similar HTTP requests clustering

### DISCUSSION THREAD

**Discussion by anquanbiji** [projectdiscovery/nuclei#2085] [General]
### Please describe your feature request:
Hope to merger the requests when the requests of  templates are the same. 

### Describe the use case of this feature:
For tech,  send as few requests as possible to get more useful information, avoid being banned by WAF.  
For example, the request of  wondercms-detect  and tech-detect  is same, so send one request to get the result for two templates I think. 

Thanks for your answer.

**Comment by ehsandeep** [comment]
@anquanbiji this is built-in and already supported, for example, here you can see running 67 templates and only making a single HTTP request as all the templates share the same request.

```console
nuclei -t takeovers/ -u https://example.com

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.7.1

		projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Using Nuclei Engine 2.7.1 (latest)
[INF] Using Nuclei Templates 9.0.3 (latest)
[INF] Templates added in last update: 47
[INF] Templates loaded for scan: 67
[INF] Templates clustered: 67 (Reduced 66 HTTP Requests)
[INF] No results found. Better luck next time!
```

The example you shared about `wondercms-detect` and `tech-detect` is not exactly same, as [tech-detect](https://github.com/projectdiscovery/nuclei-templates/blob/55d54b57c9bc0d797a28add055d939a275d23554/technologies/tech-detect.yaml) template follows HTTP redirects, and [wondercms-detect](https://github.com/projectdiscovery/nuclei-templates/blob/55d54b57c9bc0d797a28add055d939a275d23554/technologies/wondercms-detect.yaml) does not, request clustering works in case of everything is same.

**Reply by anquanbiji** [reply]
Cool, I see. Thanks for your help.
